hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -270,28 +300,32 @@ private static Bytes resultFrom(final ResponseCodeEnum status) {
 		return UInt256.valueOf(status.getNumber());
 	}
 
+	private void decodeInput(Bytes input) {
+		this.transactionBody = TransactionBody.newBuilder();
+		try {
+			this.transactionBody = this.precompile.body(input);
+		} catch (Exception e) {
+			log.warn("Internal precompile failure", e);
+			throw new InvalidTransactionException("Cannot decode precompile input", FAIL_INVALID);
+		}
+	}
+
 	@SuppressWarnings("rawtypes")
-	private Bytes computeInternal(
-			final MessageFrame frame,
-			final Bytes input,
-			final Precompile precompile
+	protected Bytes computeInternal(
+			final MessageFrame frame
 	) {
-		final var contract = frame.getContractAddress();
-		final var recipient = frame.getRecipientAddress();
 		final var updater = (AbstractLedgerWorldUpdater) frame.getWorldUpdater();
 		final var ledgers = updater.wrappedTrackingLedgers();
 
-		TransactionBody.Builder synthBody = TransactionBody.newBuilder();
 		Bytes result = SUCCESS_RESULT;
 		ExpirableTxnRecord.Builder childRecord;
 		try {
-			synthBody = precompile.body(input);
-			childRecord = precompile.run(recipient, contract, ledgers);
+			childRecord = this.precompile.run(frame, ledgers);
 
-			if (precompile instanceof MintPrecompile && childRecord.getReceiptBuilder() != null) {
+			if (this.precompile instanceof MintPrecompile && childRecord.getReceiptBuilder() != null) {
```

Comment:
These `instanceof` checks feel like they should be a method `calculateResult` on `com.hedera.services.store.contracts.precompile.HTSPrecompiledContract.Precompile`, where the parent function returns `SUCCESS_RESULT`

Reply:
Makes sense to me. Fixed.